packages/experience/src/Providers/UserInteractionContextProvider/UserInteractionContext.tsx

```diff
@@ -2,20 +2,20 @@ import { type SsoConnectorMetadata } from '@logto/schemas';
 import { noop } from '@silverhand/essentials';
 import { createContext } from 'react';
 
-export type SingleSignOnContextType = {
+export type UserInteractionContextType = {
   // All the enabled sso connectors
   availableSsoConnectorsMap: Map<string, SsoConnectorMetadata>;
-  email?: string;
-  setEmail: React.Dispatch<React.SetStateAction<string | undefined>>;
+  ssoEmail?: string;
```

Comment:
The idea of extending the scope of this ContextPovider is great.  I had a similar thought, how about we maintain a single email field within an interaction context? Technically users will either sign-in using an email identifier or through SSO. Should the input email be a shared value across all these sign-in forms?

Reply:
Great, issue added (LOG-9519).
I will do a refactor after the caching identifier input value feature is done.